gensim/models/ldamodel.py

```diff
@@ -836,49 +838,16 @@ def top_topics(self, corpus, num_words=20):
         for topic in self.state.get_lambda():
             topic = topic / topic.sum()  # normalize to probability distribution
             bestn = matutils.argsort(topic, topn=num_words, reverse=True)
-            topics.append(bestn)
+            topics.append([self.id2word[id] for id in bestn]) # Coherence Model requires a list of tokenized topics
             beststr = [(topic[id], self.id2word[id]) for id in bestn]
             str_topics.append(beststr)
 
-        # top_ids are limited to every topics top words. should not exceed the
-        # vocabulary size.
-        top_ids = set(chain.from_iterable(topics))
-
-        # create a document occurence sparse matrix for each word
-        doc_word_list = {}
-        for id in top_ids:
-            id_list = set()
-            for n, document in enumerate(corpus):
-                if id in frozenset(x[0] for x in document):
-                    id_list.add(n)
-
-            doc_word_list[id] = id_list
+        cm  = CoherenceModel(topics=topics, corpus=corpus, dictionary=self.id2word, coherence='u_mass', topn=num_words, aggregated=False)
```

Comment:
Yes it can be. But, I havent made changes for coherence metrics apart from `u_mass` to generate non-aggregated coherence scores. Also since other coherence metrics require few mandatory parameters to be passed in `CoherenceModel`, will have to change that. Is it required in this PR?